hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/handlers/TssUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.tss.handlers;
+
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.hapi.services.auxiliary.tss.TssMessageTransactionBody;
+import com.hedera.node.app.tss.api.TssLibrary;
+import com.hedera.node.app.tss.api.TssMessage;
+import com.hedera.node.app.tss.api.TssParticipantDirectory;
+import com.hedera.node.app.tss.pairings.FakeFieldElement;
+import com.hedera.node.app.tss.pairings.FakeGroupElement;
+import com.hedera.node.app.tss.pairings.PairingPrivateKey;
+import com.hedera.node.app.tss.pairings.PairingPublicKey;
+import com.hedera.node.app.tss.pairings.SignatureSchema;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.math.BigInteger;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class TssUtils {
+    /**
+     * Compute the TSS participant directory from the roster.
+     * @param roster the roster
+     * @param maxSharesPerNode the maximum number of shares per node
+     * @param selfNodeId the node ID of the current node
+     * @return the TSS participant directory
+     */
+    public static TssParticipantDirectory computeTssParticipantDirectory(
+            @NonNull final Roster roster, final long maxSharesPerNode, final int selfNodeId) {
+        final var computedShares = computeNodeShares(roster.rosterEntries(), maxSharesPerNode);
+        final var totalShares =
+                computedShares.values().stream().mapToLong(Long::longValue).sum();
+        final var threshold = (int) (totalShares + 2) / 2;
+
+        final var builder = TssParticipantDirectory.createBuilder().withThreshold(threshold);
+        // FUTURE: This private key must be loaded from disk
```

Comment:
I'm making the TSS key available in the `Node` object in my PR 👍

Reply:
Thanks @mhess-swl !